### PR TITLE
feat: update packages to make publish smaller

### PR DESCRIPTION
**Pull request by leangseu-edx**
- update dependency and peer dependency for the project.
- replace `@edx/frontend-plugin-framework` with `@openedx/frontend-plugin-framework`
- update example to use the latest compiled code whether is from `npm start` or `npm run build`
- add script to run example with `npm run start:example`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,38 +35,43 @@
...
     "glob": "7.2.3",
     "husky": "7.0.4",
     "jest": "29.7.0",
+    "prop-types": "^15.8.0",
+    "react": "^17.0.0",
+    "react-dom": "^17.0.0",
+    "react-error-boundary": "^4.0.11",
+    "react-redux": "7.2.9",
+    "react-router": "6.22.2",
+    "react-router-dom": "6.22.2",
```
Comment (MaxFrank13) [inline package.json:60]: I'm wondering if maybe we also can remove the `react-router` packages from the root `package.json`. They are only used in example apps as far as I know.

Comment (arbrandes) [reply package.json:60]: +1

**Comment by leangseu-edx** [comment]
I just realized that I do not have permission to merge this pr. @jsnwesson can you merge after review it?

**Comment by jsnwesson** [comment]
Hey @leangseu-edx ! Thanks for looking into this, although I noticed that when I tried running the "host" example app, it wasn't able to render the plugins. Having compared it to the `master` branch, I can confirm that some dependency in this PR is causing those plugins to go missing. Would you be able to look into it? 

Screenshot 2024-03-26 at 8 55 25 AM

**Comment by leangseu-edx** [comment]
> Hey @leangseu-edx ! Thanks for looking into this, although I noticed that when I tried running the "host" example app, it wasn't able to render the plugins. Having compared it to the `master` branch, I can confirm that some dependency in this PR is causing those plugins to go missing. Would you be able to look into it?

I found the issue. It seems like 
```
    "@edx/brand": "npm:@openedx/brand-openedx@^1.2.2",
    "@edx/frontend-component-footer": "13.0.3",
    "@edx/frontend-component-header": "5.0.2",
```
are required at the root of the project. I don't know why it seems necessary.

Additional file changed are just replacing `@edx/frontend-plugin-framework` with `@openedx/frontend-plugin-framework`.

**Comment by jsnwesson** [comment]
@leangseu-edx no apologies needed, this is all greatly appreciated given the time you've put into making the experience cleaner! 
One thing to note is that there looks to be a mix up of port numbers in the changes you made so far, but there's also other parts of the example files that likely need to be synced with the correct port number, as I wasn't able to see any of the iFrame-based plugins in the host MFE. 

To make sure the terms are clear: 
- "Host MFE" AKA the `example` app
  - Original PORT was 8080 according to the JS-config values, `BASE_URL` and `PORT`
  - Now looks to be 8081 according to `webpack.dev.config.js`
  
- "Child MFE" AKA the `example-plugin-app`
  - Original PORT was 8081 according to the JS-config values `BASE_URL` and `PORT` and the customization in `webpack.dev.config.js`
  - Now looks to be 8082 according to `webpack.dev.config.js`
  
Oddly enough, I was able to see the Host MFE by going to `localhost:8082`, which according to your changes should've rendered the Child MFE instead. Not sure why that is. 
Additionally, I noticed in the README "Getting Started" section that devs should go to `localhost:8080` to see the example app, but perhaps it was meant to point to whatever the port is for Host MFE?

**Review by jsnwesson — APPROVED** [review]

**Comment by jsnwesson** [comment]
Sweeet thanks for the fix! Approved and greatly appreciated. Thanks again @leangseu-edx !

**Pull request merged**